processing/src/main/java/io/druid/query/filter/IntervalDimFilter.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.filter;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.RangeSet;
+import com.metamx.common.StringUtils;
+import io.druid.common.utils.JodaUtils;
+import io.druid.query.extraction.ExtractionFn;
+import io.druid.query.ordering.StringComparators;
+import org.joda.time.Interval;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class IntervalDimFilter implements DimFilter
+{
+  private final List<Interval> intervals;
+  private final String dimension;
+  private final ExtractionFn extractionFn;
+  private final List<DimFilter> boundDimFilters;
+
+  @JsonCreator
+  public IntervalDimFilter(
+      @JsonProperty("dimension") String dimension,
+      @JsonProperty("intervals") List<Interval> intervals,
+      @JsonProperty("extractionFn") ExtractionFn extractionFn
+  )
+  {
+    Preconditions.checkNotNull(dimension, "dimension can not be null");
+    Preconditions.checkNotNull(intervals, "intervals can not be null");
+    Preconditions.checkArgument(intervals.size() > 0, "must specify at least one interval");
+    this.dimension = dimension;
+    this.intervals = Collections.unmodifiableList(JodaUtils.condenseIntervals(intervals));
+    this.extractionFn = extractionFn;
+    this.boundDimFilters = makeBoundDimFilters();
+  }
+
+  @JsonProperty
+  public String getDimension()
+  {
+    return dimension;
+  }
+
+  @JsonProperty
+  public List<Interval> getIntervals()
+  {
+    return intervals;
+  }
+
+  @JsonProperty
+  public ExtractionFn getExtractionFn()
+  {
+    return extractionFn;
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    byte[] dimensionBytes = StringUtils.toUtf8(dimension);
+
+    final byte[][] intervalsBytes = new byte[intervals.size()][];
+    int intervalsBytesSize = 0;
+    int index = 0;
+    for (Interval interval : intervals) {
```

Comment:
@gianm the filter now stores a list of start/end pairs for cache-key and filter conversion purposes